junit5-jupiter-starter-bazel/bazeljunit5/src/main/java/com/flexport/bazeljunit5/BazelJUnit5ConsoleLauncher.java

```diff
@@ -0,0 +1,339 @@
+    // environment variable would simply be `foo`. Here are some examples of what it might look
+    // like in practice: test.package.TestClass#testMethod, test.package.TestClass, test.package
+
+    List<String> newArgs = new ArrayList<>();
+    newArgs.addAll(filterOptions(args, Arrays.asList(SELECT_PACKAGE)));
+    newArgs.addAll(parseOptions(testBridgeTestOnly));
+
+    return newArgs;
+  }
+
+  private static List<String> transformArgsForXmlOutputFile(
+      List<String> args, String xmlOutputFile) {
+
+    if (xmlOutputFile == null || xmlOutputFile.isEmpty()) {
+      return args;
+    }
+
+    // The XML output reporting file is set in the XML_OUTPUT_FILE environment variable by JUnit4.
+    // And JUnit5 will get the reports directory from `--reports-dir`.
+
+    List<String> newArgs = new ArrayList<>(args);
+    newArgs.add("--reports-dir=" + Paths.get(xmlOutputFile).getParent().toString());
+
+    return newArgs;
+  }
+
+  /**
+   * Parse JUnit5 option from env.TESTBRIDGE_TEST_ONLY.
+   *
+   * <p>test.package: --select-package=test.package
+   *
+   * <p>test.package.TestClass: --select-class=test.package.TestClass
+   *
+   * <p>test.package.TestClass#testMethod: --select-method=test.package.TestClass#testMethod
+   *
+   * @param testBridgeTestOnly env.TESTBRIDGE_TEST_ONLY
+   * @return option
+   */
+  private static List<String> parseOptions(String testBridgeTestOnly) {
+    // transform env.TESTBRIDGE_TEST_ONLY
+
+    // test.package.TestClass#
+    if (testBridgeTestOnly.endsWith("#") || testBridgeTestOnly.endsWith("$")) {
+      testBridgeTestOnly = testBridgeTestOnly.substring(0, testBridgeTestOnly.length() - 1);
+    }
+
+    if (testBridgeTestOnly.contains("#")) {
+      // There could be multiple classes in TESTBRIDGE_TEST_ONLY which are separated by $|
+      String[] splits = testBridgeTestOnly.split("\\$\\|");
+      HashSet<String> methodNames = new HashSet<>();
+      HashSet<String> classNames = new HashSet<>();
+
+      for (String split : splits) {
+        String[] parts = split.split("#");
+        String className = parts[0];
+        classNames.add(className);
+        methodNames.addAll(getMethodNames(parts[1]));
+      }
+
+      List<String> parsedOptions = new ArrayList<>();
+
+      for (String className : classNames) {
+        Class<?> klass;
+        try {
+          klass = Class.forName(className);
+        } catch (ClassNotFoundException e) {
+          throw new IllegalStateException(e);
+        }
+
+        parsedOptions.addAll(Arrays.stream(klass.getDeclaredMethods())
+            .filter(method -> methodNames.contains(method.getName()))
+            .map(
+                method ->
+                    SELECT_METHOD + "=" + ReflectionUtils
+                        .getFullyQualifiedMethodName(klass, method))
+            .collect(Collectors.toList()));
+      }
+
+      return parsedOptions;
+    }
+
+    try {
+      Class.forName(testBridgeTestOnly);
```

Comment:
`--test_filter=package.class.method` is also valid. In fact, my IDE, IntelliJ IDEA 2020.2, is automatically generating this form for Kotlin test code.

Probably need more logic to handle this

Reply:
Hum, when started from within IntelliJ IDEA, this doesn't seem to work for nested classes.